src/time.cpp

```diff
@@ -69,6 +71,27 @@ namespace g3 {
    /// * format string must conform to std::put_time
    /// This is similar to std::put_time(std::localtime(std::time_t*), time_format.c_str());
 
+   std::string localtime_formatted(const timespec &time_snapshot, const std::string &time_format) {
+      ushort fcount = 0;
+      for(auto found_ptr = time_format.c_str(); (found_ptr = strstr(found_ptr, "%f")) != nullptr; found_ptr++, fcount++);
+      const size_t format_size = time_format.length() + fcount * 7 + 1;  // len(format) + num(nanosec size - len("%f")) + zero
```

Comment:
My target platform is cheap ARM boards (Pi and alike) with Linux RT kernel. As far as I know it uses free running counter at 1 MHh. Which means precision granularity is microsec and there is no point in nanosec for me, but it might be different for other g3log users.